mne/io/eeglab/eeglab.py

```diff
@@ -260,6 +330,284 @@ def read_epochs_eeglab(input_fname, events=None, event_id=None, montage=None,
+            mapping[key] = _bunchify(value)
+        return _bunch_wrapper(name, **mapping)
+    elif isinstance(mapping, list):
+        return [_bunchify(item) for item in mapping]
+    return mapping
+
+
+def _bunch_wrapper(name, **kwargs):
+    """Convert mappings to Bunches."""
+    return Bunch(**kwargs)
+
+
+def _bunch_data_2_strs(bunch_data, field, lower=True):
+    """Convert ascii strings to human-readable strings.
+
+    Takes string values stored as ascii values in numpy arrays
+    in bunch objects and returns list of human-readable strings
+    """
+    str_list = [''.join([chr(x) for x in curr_label.__dict__[field]]).strip()
+                for curr_label in bunch_data]
+    return str_list
+
+
+def _bunch_str_conversions(bunch_data, str_conversion_fields):
+    """Convert selected fields.
+
+    Fields are from bunch object from 1D array
+    of ascii values of chars.
+    """
+    for curr_field in str_conversion_fields:
+        c1 = (len(bunch_data) > 0)
+        c2 = (c1 and (curr_field in bunch_data[0]))
+        c3 = (c2 and
+              not isinstance(bunch_data[0].__dict__[curr_field], str))
+        if c3:
+            str_data = _bunch_data_2_strs(bunch_data, curr_field)
+            for ctr, curr_str in enumerate(str_data):
+                bunch_data[ctr].__dict__[curr_field] = curr_str
+    return bunch_data
+
+
+def _bunch_derefs(orig, bunch_data, deref_fields):
+    """Dereference h5py.h5r.Reference objects.
+
+    Ensures that each field of bunch object with dereferenced
+    objects stores them in a list, even if that list has only
+    1 element.
+    """
+    try:
+        import h5py
+    except ImportError:
+        raise RuntimeError('Reading v7+ MATLAB format .set',
+                           'requires h5py, which could not',
+                           'be imported')
+
+    for curr_field in deref_fields:
+        bd = bunch_data[0].__dict__[curr_field]
+        c1 = (len(bunch_data) > 0)
+        c2 = (curr_field in bunch_data[0])
+        c3 = (len(bd) > 0)
+        c4 = (isinstance(bd[0], h5py.h5r.Reference))
+        if (c1 and c2 and c3 and c4):
+            for ctr in range(len(bunch_data)):
+                bd = bunch_data[ctr].__dict__[curr_field]
+                try:
+                    # Ensure bunch_data[ctr].__dict__[curr_field] is iterable
+                    # before attempting to iterate over it
+                    iter(bd)
+                except TypeError:
+                    deref = [orig[bd].value.flatten()]
+                else:
+                    deref = [orig[x].value.flatten() for x in bd]
+
+                bunch_data[ctr].__dict__[curr_field] = deref
+
+    return bunch_data
+
+
+def _get_hdf_eeg_data(input_fname):
+
+    try:
+        import h5py
+    except ImportError:
+        raise RuntimeError('Reading v7+ MATLAB format .set',
+                           'requires h5py, which could not',
+                           'be imported')
```

Comment:
I think that this is the only place this check should be done, i.e. when you try to first load the data in HDF5 format. All the others should be removed / removable.